lib/internal/process/next_tick.js

```diff
@@ -246,7 +246,7 @@ function setupNextTick() {
     if (typeof callback !== 'function')
       throw new errors.TypeError('ERR_INVALID_CALLBACK');
 
-    if (process._exiting)
+    if (process.isExiting)
```

Comment:
I would expect this to introduce a performance regression (getter that calls into the C++ runtime.)  Have you run next-tick benchmarks?

Reply:
Perhaps rather than a getter here, a public read-only property should be added *only* when we are exiting... that is.. `setIsExiting()` could do something like...

```js
Object.defineProperty(process, 'exiting', { configurable: false, enumerable: true, value: true });
```

This should largely avoid the perf regression on calling the getter multiple times.

Reply:
@jasnell If it weren't there prior to exiting, then it could be set by userland code long before exiting, causing the problems I was trying to solve.

Maybe a getter to something in `internal/process`? That way it still would be userland tamper-proof, but wouldn't dive into C++.

Reply:
That would certainly be better but the getter still comes at a cost.... hmm will have to think about it a bit more.

Reply:
`nextTick` could just use the actual value on `internal/process`, rather than the getter (which would be only for public use).

Reply:
@bengl if I understand you correct you want to use a plain symbol in `internal/process` plus the `isExiting` property as getter that exposes that symbol?